### PR TITLE
add logic to skip empty labels

### PR DESCRIPTION
**Pull request by kaimikacolin**

### CODE REVIEW

**Comment by kaimikacolin** [comment]
Should "All" and "Unlabeled" be skipped as well?

File: web/views.py [hunk 1]
```diff
@@ -174,10 +174,10 @@ def get_labels(request):
         labels = Label.objects.all().order_by('name')
     else:
         rc = get_object_or_404(ImageCollection, name=collection_name)
-        labels = rc.labels()
+        labels = rc.labels(check_if_has_winning=True) # This is currently our switch to enable/disable skipping empty labels
```
Comment (kaimikacolin) [inline web/views.py:177]: This is currently the only way to enable/disable the functionality.

File: core/models.py [hunk 2]
```diff
@@ -209,9 +209,17 @@ class ImageCollection(models.Model):
...
     rois = models.ManyToManyField(ROI, related_name="collections")
 
-    def labels(self):
+    def labels(self, check_if_has_winning=False):
         label_ids = [i['annotations__label'] for i in self.rois.values('annotations__label').distinct()]
-        return sorted(Label.objects.in_bulk(label_ids).values(), key=lambda label: label.name.lower())
+        labels = sorted(Label.objects.in_bulk(label_ids).values(), key=lambda label: label.name.lower())
+
+        if check_if_has_winning:
+            winning_label_ids = [i['winning_annotation__label'] for i in self.rois.values('winning_annotation__label').distinct()]
```
Comment (kaimikacolin) [inline core/models.py:217]: I did not use exists() here, but I think it should still be quite speedy

File: core/models.py [hunk 3]
```diff
@@ -209,9 +209,17 @@ class ImageCollection(models.Model):
...
+            ret = [{'label_name': i.name, 'has_winning': i.id in winning_label_ids} for i in labels]
+        else:
+            ret = [{'label_name': i.name, 'has_winning': True} for i in labels]
+
+        return ret
```
Comment (kaimikacolin) [inline core/models.py:222]: I did not use exists() here, but I think it should still be quite speedy

**Comment by joefutrelle** [comment]
This may interact with #150

**Pull request merged**